client/src/rubin/nublado/client/testing/_jupyter.py

```diff
@@ -414,9 +446,17 @@ def get_content(self, request: Request) -> Response:
         user = self._get_user_from_headers(request)
         if user is None:
             return Response(403, request=request)
+        if redirect := self._maybe_redirect(request, user):
+            return Response(
+                302, request=request, headers={"Location": redirect}
+            )
```

Comment:
All this repetitive boilerplate is making me nostalgiac for Go.

Yeah, I get that since we have to return from the function if this request generates a redirect, there's no better way to do this.

A way to shorten this and reduce the duplication might be to convert _maybe_redirect into something that raised an Exception that had the redirect as a field, and then put everything here inside a try: if you catch that Exception, return the redirect.  That might be a lot less clear though.